### PR TITLE
Refactor special track elements list part 1

### PR DESCRIPTION
**Pull request by spacek531**
- Tweaks how special track elements are selected in the special elements dropdown
- Makes large curves (diagonal curves) a track flag instead of a ride flag for consistency
- Makes flat ride bases its own track flag instead of flat track, to prevent them showing up in special elements
- Adds missing data in TrackData.cpp
- Fixes https://github.com/OpenRCT2/OpenRCT2/issues/19900
- Prerequisite for https://github.com/OpenRCT2/OpenRCT2/pull/19919
- Bumped network version just in case (is it even needed?)

Tested building these rides, all work as expected:
- RMC raptor coaster
- Mine train coaster
- merry go round
- car ride

### CODE REVIEW

File: src/openrct2/ride/TrackData.cpp [hunk 1]
```diff
@@ -4847,13 +4856,14 @@ static constexpr money64 TrackPricing[TrackElemType::Count] = {
...
-    294912, // TrackElemType::RightEighthToOrthogonalUp25  
-    294912, // TrackElemType::LeftEighthToOrthogonalDown25 
+    294912, // TrackElemType::LeftEighthToDiagUp25
+    294912, // TrackElemType::RightEighthToDiagUp25
+    294912, // TrackElemType::LeftEighthToDiagDown25
+    294912, // TrackElemType::RightEighthToDiagDown25
+    294912, // TrackElemType::LeftEighthToOrthogonalUp25
+    294912, // TrackElemType::RightEighthToOrthogonalUp25
+    294912, // TrackElemType::LeftEighthToOrthogonalDown25
+    294912, // TrackElemType::RightEighthToOrthogonalDown25
```
Comment (spacek531) [inline src/openrct2/ride/TrackData.cpp:4866]: This data was missing

File: src/openrct2/ride/TrackData.cpp [hunk 2]
```diff
@@ -4474,6 +4478,10 @@ static constexpr track_type_t AlternativeTrackTypes[TrackElemType::Count] = {
     TrackElemType::None,
     TrackElemType::None,
     TrackElemType::None,
+    TrackElemType::None,
+    TrackElemType::None,
+    TrackElemType::None,
+    TrackElemType::None,
```
Comment (spacek531) [inline src/openrct2/ride/TrackData.cpp:4484]: These data were missing

File: src/openrct2/ride/TrackData.cpp [hunk 3]
```diff
@@ -7694,13 +7711,13 @@ static constexpr const StringId RideConfigurationStringIds[] = {
     STR_HELIX_UP_RIGHT,                // 107
     STR_HELIX_DOWN_LEFT,               // 108
     STR_HELIX_DOWN_RIGHT,              // 109
-    STR_BASE_SIZE_2_X_2,               // 110
-    STR_BASE_SIZE_4_X_4,               // 111
+    STR_EMPTY,                         // TrackElemType::Up25LeftBanked
+    STR_EMPTY,                         // TrackElemType::Up25RighBanked
```
Comment (spacek531) [inline src/openrct2/ride/TrackData.cpp:7715]: These strings were here as holdovers from before TrackElemType

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 4]
```diff
@@ -2110,7 +2110,7 @@ class RideConstructionWindow final : public Window
             const auto& ted = GetTrackElementDescriptor(trackType);
             int32_t trackTypeCategory = ted.Definition.type;
 
-            if (trackTypeCategory == TRACK_NONE)
+            if (ted.Description == STR_EMPTY)
```
Comment (spacek531) [inline src/openrct2-ui/windows/RideConstruction.cpp:2113]: Only special track elements have strings, AFAIK, and only those special elements will show up in the special elements dropdown.

File: src/openrct2/ride/TrackData.cpp [hunk 5]
```diff
@@ -5914,10 +5927,11 @@ static constexpr uint8_t TrackSequenceElementAllowedWallEdges[TrackElemType::Cou
     {      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0 }, // TrackElemType::LeftEighthBankToOrthogonalUp25         
     {      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0 }, // TrackElemType::RightEighthBankToOrthogonalDown25      
     {      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0,      0 }, // TrackElemType::LeftEighthBankToOrthogonalDown25       
-};//TODO
```
Comment (spacek531) [inline src/openrct2/ride/TrackData.cpp:5917]: Do you have any need for these TODO reminders? @X123M3-256

Comment (X123M3-256) [reply src/openrct2/ride/TrackData.cpp:5917]: Those were used to mark arrays that had been filled with dummy data for testing. If they were left in, that probably means I forgot to go back and put real data there so I'll have to check.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**